app/uk/gov/hmrc/apiplatform/modules/uplift/controllers/UpliftJourneyController.scala

```diff
@@ -190,6 +212,36 @@ class UpliftJourneyController @Inject() (
     successful(Ok(beforeYouStartView(sandboxAppId)))
   }
 
+  def agreeNewTermsOfUse(appId: ApplicationId): Action[AnyContent] = whenTeamMemberOnApp(appId) { implicit request =>
+    lazy val showSubmission = (s: Submission) =>
+      if (request.role.isAdministrator) {
+        if (s.status.isAnsweredCompletely) {
+          Redirect(uk.gov.hmrc.apiplatform.modules.submissions.controllers.routes.CheckAnswersController.checkAnswersPage(appId))
+        } else {
+          Redirect(uk.gov.hmrc.apiplatform.modules.submissions.controllers.routes.ProdCredsChecklistController.productionCredentialsChecklistPage(appId))
+        }
+      } else {
+        Ok(unauthorisedAppDetailsView(request.application.name, request.application.adminEmails))
+      }
+
+    lazy val showBeforeYouStart =
+      if (request.role.isAdministrator) {
+        Ok(beforeYouStartView(appId))
+      } else {
+        Ok(unauthorisedAppDetailsView(request.application.name, request.application.adminEmails))
+      }
+
+    val x =
```

Comment:
I would remove the value assignment here and fold on the result below

Reply:
Fixed